pennylane/workflow/_setup_transform_program.py

```diff
@@ -119,7 +119,7 @@ def _setup_transform_program(
         or resolved_execution_config.interface is Interface.NUMPY
         or resolved_execution_config.gradient_method == "backprop"
         or (
-            getattr(device, "short_name", "") == "default.mixed"
+            isinstance(device, qml.devices.DefaultMixed)
             and resolved_execution_config.gradient_method is None
         )
```

Comment:
Can we get rid of this patch now that `DefaultMixed` follows the new device interface? Instead, we can just use the `convert_to_numpy` configuration and just follow the standard device interface.

Reply:
Trying at https://github.com/PennyLaneAI/pennylane/pull/6684/commits/0f25a36655d69190a25a493970df8d0daa3a2d15

Reply:
It looks all good!